src/connection.js

```diff
@@ -136,30 +136,33 @@ module.exports = class Connection extends EventEmitter {
             }
 
             if (should_reconnect) {
+                var reconnect_wait = that.calculateExponentialBackoff();
+                
                 that.reconnect_attempts++;
                 that.emit('reconnecting', {
                     attempt: that.reconnect_attempts,
                     max_retries: that.auto_reconnect_max_retries,
-                    wait: that.auto_reconnect_wait
+                    wait: reconnect_wait
                 });
+                
+                that.debugOut('Scheduling reconnect');
+                that.setTimeout(function() {
```

Comment:
Since you moved it anyway: 
```suggestion
                that.setTimeout(() => that.connect, reconnect_wait);
```